File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        if len(nums) == len(set(nums)):
            return False
        store ={}

        for i in range(len(nums)):
            if nums[i] in store and i- store[nums[i]] <= k:
                return True
            store[nums[i]] = i
        return False

File: 219-contains-duplicate-ii/contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int> store;
        for(int i=0;i<nums.size();i++){
            if(store.contains(nums[i]) && i - store[nums[i]] <= k) return true;
            store[nums[i]] = i;
        }
        return false;
    }
};